workflow/controller/controller.go

```diff
@@ -915,7 +915,12 @@ func (wfc *WorkflowController) tweakListRequestListOptions(options *metav1.ListO
 	options.LabelSelector = labelSelector.String()
 	// `ResourceVersion=0` does not honor the `limit` in API calls, which results in making significant List calls
 	// without `limit`. For details, see https://github.com/argoproj/argo-workflows/pull/11343
-	options.ResourceVersion = ""
+	if options.ResourceVersion == "0" {
+		options.ResourceVersion = ""
+	}
```

Comment:
The recorded LastResourceVersion cannot be overwritten.  see https://github.com/kubernetes/client-go/blob/ee1a5aaf793a9ace9c433f5fb26a19058ed5f37c/tools/cache/reflector.go#L574